src/Microsoft.DotNet.Interactive.Js/src/dotnet-interactive/client-side-kernel.ts

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+import { ClientSideKernelInvocationContext } from "./client-side-kernel-invocation-context";
+import { DisposableSubscription, KernelEventEnvelopeObserver, KernelCommandEnvelope, KernelTransport, Disposable, KernelEventEnvelope } from "./contracts";
+import { Kernel, KernelInvocationContext } from "./dotnet-interactive-interfaces";
+import { TokenGenerator } from "./tokenGenerator";
+
+export class ClientSideKernel implements Kernel {
+    private _commandHandlers: { [commandType: string]: (envelope: KernelCommandEnvelope, context: KernelInvocationContext) => Promise<void> } = {};
```

Comment:
a command handler signature could should just deal with kernelcommands

Reply:
Are you referring to the envelope versus command, @colombod?